### PR TITLE
Implemented L0 tests

### PR DESCRIPTION
**Pull request by kuvinodms**
Implemented L0 tests

### CODE REVIEW

File: Tasks/PyPIPublisher/Tests/L0PythonAvailability.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import ma = require('vsts-task-lib/mock-answer');
+import tmrm = require('vsts-task-lib/mock-run');
+import path = require('path');
+const util = require('util')
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0PythonAvailability.ts]: use same import command for all modules.

File: Tasks/PyPIPublisher/Tests/L0PythonAvailability.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0PythonAvailability.ts]: remove this empty line

File: Tasks/PyPIPublisher/Tests/L0PythonAvailability.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+var os = require('os');
+var homedir = os.homedir();
+let pypircFilePath: string = path.join(homedir, ".pypirc").replace(/\\/g, "\\\\");
+
+let publisher = path.join(__dirname, '..', 'publisher.js');
+let tmr: tmrm.TaskMockRunner = new tmrm.TaskMockRunner(publisher);
+tmr.setInput('serviceEndpoint', 'MyTestEndpoint');
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0PythonAvailability.ts]: Get test service endpoint name from process env

File: Tasks/PyPIPublisher/Tests/L0.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        //process.env['TASK_TEST_TRACE'] = 1; 
+
+        //setup endpoint
+        process.env["ENDPOINT_AUTH_MyTestEndpoint"] = "{\"parameters\":{\"username\":\"username\", \"password\":\"password\"},\"scheme\":\"usernamepassword\"}";
+        process.env["ENDPOINT_URL_MyTestEndpoint"] = "https://example/test";
+        process.env["ENDPOINT_AUTH_PARAMETER_MyTestEndpoint_USERNAME"] = "username";
+        process.env["ENDPOINT_AUTH_PARAMETER_MyTestEndpoint_PASSWORD"] = "password";
+    });
+
+    it('Python tool should be present', (done: MochaDone) => {
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0.ts]: better test name

File: Tasks/PyPIPublisher/Tests/L0.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        this.timeout(1000);
+
+        let tp = path.join(__dirname, 'L0PythonAvailability.js');
+        let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        tr.run();
+        assert(tr.succeeded, 'task should have succeeded');
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0.ts]: Add more assertions regarding different actions which are performed.

File: Tasks/PyPIPublisher/Tests/L0.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import * as path from 'path';
+
+describe('PyPI Publisher', function () {
+    
+    before(() => {
+        //Enable this for output
+        //process.env['TASK_TEST_TRACE'] = 1; 
+
+        //setup endpoint
+        process.env["ENDPOINT_AUTH_MyTestEndpoint"] = "{\"parameters\":{\"username\":\"username\", \"password\":\"password\"},\"scheme\":\"usernamepassword\"}";
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0.ts]: These might not be available in test file. please move if needed.

File: Tasks/PyPIPublisher/Tests/L0.ts (hunk 4 above)
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/Tests/L0.ts]: Add failure test cases as well

File: Tasks/PyPIPublisher/publisher.ts [hunk 7]
```diff
@@ -19,27 +19,26 @@ var password = tl.getEndpointAuthorizationParameter(serviceEndpointId, 'password
...
+        await executePythonTool(util.format("-m twine upload %s", uploadArtifactsPath));
     }
     catch(err){
-        tl.setResult(tl.TaskResult.Failed, err);
+        tl.setResult(tl.TaskResult.Failed, "Upload Failed");
     }
     finally{
-        tl.setResult(tl.TaskResult.Succeeded, "Upload Successful");
         //Delete .pypirc file
         tl.rmRF(pypircFilePath);
```
Comment (bishal-pdMSFT) [inline Tasks/PyPIPublisher/publisher.ts]: you can log a warning in case delete fails

File: Tasks/PyPIPublisher/publisher.ts [hunk 8]
```diff
@@ -19,27 +17,27 @@ var password = tl.getEndpointAuthorizationParameter(serviceEndpointId, 'password
...
-        await executePythonTool(util.format("%s sdist --dist-dir %s bdist_wheel", setupFilePath, distDirectoryPath));    
-        await executePythonTool(util.format("-m twine upload %s", uploadArtifactsPath));    
+        tl.cd(workingDirectory);
+        await executePythonTool("-m pip install twine --user");
+        await executePythonTool("setup.py sdist");
+        await executePythonTool("-m twine upload dist/*");
     }
     catch(err){
-        tl.setResult(tl.TaskResult.Failed, err);
+        tl.setResult(tl.TaskResult.Failed, "Upload Failed");
```
Comment (sachinma) [inline Tasks/PyPIPublisher/publisher.ts]: This should be localized

Comment (sachinma) [reply Tasks/PyPIPublisher/publisher.ts]: Are you hiding the original error

**Pull request merged**